src/app/runtime_genesis_ledger/runtime_genesis_ledger.ml

```diff
@@ -145,6 +145,9 @@ let main ~(constraint_constants : Genesis_constants.Constraint_constants.t)
   let%bind hash_json =
     generate_hash_json ~genesis_dir ledger staking_ledger next_ledger
   in
+  Ledger.close ledger ;
+  Ledger.close staking_ledger ;
+  Ledger.close next_ledger ;
```

Comment:
Doesn't this occur after the tarballs are generated? I'm wondering how this would affect the outcome of the test.

I looked at the [failing test you posted](https://buildkite.com/organizations/o-1-labs-2/pipelines/mina-end-to-end-nightlies/builds/3967/jobs/019a4d9a-491c-40f7-87b3-976b6446807e/log) and it appears that two epoch ledgers were created, both named `epoch_ledger_d88bbcc4a34910b56e7f37ef92bf3edc5ae6acd31a5fea48f364cb4cc65a8b42`. Maybe the fix in https://github.com/MinaProtocol/mina/pull/17935 is somehow not enough?